src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -511,7 +512,12 @@ private Map<Integer, Float> doANNSearch(
         FilterIdsSelector.FilterIdsSelectorType filterType = filterIdsSelector.getFilterType();
         // Now that we have the allocation, we need to readLock it
         indexAllocation.readLock();
-        indexAllocation.incRef();
+        try {
+            indexAllocation.incRef();
+        } catch (IllegalStateException e) {
+            indexAllocation.readUnlock();
+            throw new OpenSearchException("failed to create knn search when clear cache");
```

Comment:
```suggestion
            throw new OpenSearchException("failed to do kNN search when vector data structures getting evicted ");
```

@kotwanikunal , @jmazanec15 any thoughts on the exception message?

Reply:
I think a simple "failed to perform kNN search due to high throughput on the node" would suffice. I feel we should not expose internal details   

```suggestion
throw new OpenSearchException("failed to perform kNN search due to high throughput on the node ");
```

OR
```suggestion
 throw new OpenSearchException("failed to perform kNN search due to resource constraints on the node");
```

Reply:
> @kotwanikunal @navneet1v i prefer to
> 
> ```
>  throw new OpenSearchException("failed to do kNN search when vector data structures getting evicted ");
> ```
> 
> because this happens when do clear cache but not comes from throughput or resource constraints.

Lets got with this.